connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -143,6 +162,70 @@ public ConnectorStateInfo connectorStatus(String connName) {
                 status.workerId(), status.trace());
     }
 
+
+    @Override
+    public ConfigInfos validateConfigs(String connType, Map<String, String> connectorConfig) {
+        ConfigDef connectorConfigDef = ConnectorConfig.configDef();
+        List<ConfigValue> connectorConfigValues = connectorConfigDef.validate(connectorConfig);
+        ConfigInfos result = generateResult(connType, connectorConfigDef.configKeys(), connectorConfigValues);
+
+        if (result.errorCount() != 0) {
+            return result;
+        }
+
+        ConnectorConfig connConfig = new ConnectorConfig(connectorConfig);
+        String connName = connConfig.getString(ConnectorConfig.NAME_CONFIG);
+        Connector connector = getConnector(connType, connName);
+
+        Map<String, ConfigKey> configKeys = connector.configDef().configKeys();
+        List<ConfigValue> configValues = connector.validate(connectorConfig);
+        configKeys.putAll(connectorConfigDef.configKeys());
```

Comment:
This approach seems pretty weird. Are we modifying state in `ConfigDef` during validation? I feel like there are a few different issues with this -- it won't be thread safe, it ties state to the `ConfigDef` that shouldn't really be part of it, and it allows different config validations to get conflated. Why would we even be modifying the config keys in validation? Seems like validation should only generate `ConfigValue` objects.